ui/src/components/sermons/SermonDetails.js

```diff
@@ -84,35 +100,66 @@ const SermonDetails = (props) => {
   };
 
   const getVideoCode = () => {
-    let sermonVideoCode =
-      sermon.sermonVideoUrl.split("/")[
-        sermon.sermonVideoUrl.split("/").length - 1
-      ];
-    setSermonVideoCode(sermonVideoCode);
+    if (sermon.streamLink === '') {
+      let sermonVideoCode =
+        sermon.sermonUrl.split('/')[sermon.sermonUrl.split('/').length - 1];
+      setSermonVideoCode(`https://www.youtube.com/embed/${sermonVideoCode}`);
+    } else {
+      setSermonVideoCode(sermon.streamLink);
+    }
+  };
+
+  const randomIndex = (max) => {
+    var arr = [];
+    while (arr.length < Math.min(max, 5)) {
+      var r = Math.floor(Math.random() * max);
+      if (arr.indexOf(r) === -1) arr.push(r);
+    }
+    return arr;
   };
 
   const getRandomSermons = () => {
     let sameYearSermons = allSermons.filter(
       (random) =>
         DateTime.fromISO(sermon.datePreached).hasSame(
           DateTime.fromISO(random.datePreached),
-          "year"
+          'year'
         ) && random.sermonSeries[0].name !== sermon.sermonSeries[0].name
     );
     let randomSermons = [];
-    while (randomSermons.length < 5) {
-      randomSermons.push(
-        sameYearSermons[Math.floor(Math.random() * sameYearSermons.length)]
-      );
-      randomSermons = [...new Set(randomSermons)];
-    }
+    randomIndex(sameYearSermons.length).forEach((i) => {
+      randomSermons.push(sameYearSermons[i]);
+    });
     randomSermons = randomSermons.sort(
       (a, b) =>
         DateTime.fromISO(b.datePreached) - DateTime.fromISO(a.datePreached)
     );
     setRandomSermons(randomSermons);
   };
 
+  const refreshSermonNotes = () => {
+    setNoteId(noteId + 1);
+  };
+
+  const OnlineSection = () => {
+    return (
+      <>
+        <OnlinePageButtons />
+        <Box bgColor="#F1F1F3" p={5} borderRadius={15}>
+          <Text fontWeight="bold" color="#0628A3" fontSize="md">
+            Description:
+          </Text>
+          <Text>{sermon.sermonDesc}</Text>
+        </Box>
+        <OnlinePageTabs
+          noteId={noteId}
+          sermonNotes={sermon.sermonNotes}
+          refreshCallback={refreshSermonNotes}
```

Comment:
[NO-ACTION] no need to action on this, but this whole refreshSermonNotes should ideally be handled by OnlinePageTabs itself.
```
<OnlinePageTabs sermonNotes={sermon.sermonNotes} />
```